### PR TITLE
Add support for remote services (such as Firebase Test Lab)

### PR DESCRIPTION
**Pull request by Laimiux**
I confirmed that this works with **0.8.0**, but it doesn't work with current master.

### CODE REVIEW

**Comment by xiphirx** [comment]
What is the difference between 0.8.0 and master that is preventing this from working?

**Comment by Laimiux** [comment]
I'm running issues when trying to assemble core JAR:
```
> Task :core:compileReleaseJavaWithJavac FAILED
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:20: error: package android.support.v4.view does not exist
import android.support.v4.view.ViewCompat;
                              ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:21: error: package android.support.v4.view.accessibility does not exist
import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat;
                                            ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:22: error: package android.support.v4.view.accessibility.AccessibilityNodeInfoCompat does not exist
import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.AccessibilityActionCompat;
                                                                        ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:23: error: package android.support.v4.view.accessibility.AccessibilityNodeInfoCompat does not exist
import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.CollectionInfoCompat;
                                                                        ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:24: error: package android.support.v4.view.accessibility.AccessibilityNodeInfoCompat does not exist
import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.CollectionItemInfoCompat;
                                                                        ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:25: error: package android.support.v4.view.accessibility.AccessibilityNodeInfoCompat does not exist
import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat.RangeInfoCompat;
                                                                        ^
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:41: error: cannot find symbol
    AccessibilityNodeInfoCompat nodeInfo = AccessibilityNodeInfoCompat.obtain();
    ^
  symbol:   class AccessibilityNodeInfoCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:41: error: cannot find symbol
    AccessibilityNodeInfoCompat nodeInfo = AccessibilityNodeInfoCompat.obtain();
                                           ^
  symbol:   variable AccessibilityNodeInfoCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:46: error: cannot find symbol
      ViewCompat.onInitializeAccessibilityNodeInfo(view, nodeInfo);
      ^
  symbol:   variable ViewCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:59: error: cannot find symbol
        for (AccessibilityActionCompat action : nodeInfo.getActionList()) {
             ^
  symbol:   class AccessibilityActionCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:90: error: cannot find symbol
        CollectionInfoCompat collectionInfo = nodeInfo.getCollectionInfo();
        ^
  symbol:   class CollectionInfoCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:102: error: cannot find symbol
        CollectionItemInfoCompat collectionItemInfo = nodeInfo.getCollectionItemInfo();
        ^
  symbol:   class CollectionItemInfoCompat
  location: class AccessibilityHierarchyDumper
/Users/laimonasturauskas/androidstuffs/screenshot-tests-for-android/core/src/main/java/com/facebook/testing/screenshot/layouthierarchy/AccessibilityHierarchyDumper.java:146: error: cannot find symbol
        RangeInfoCompat rangeInfo = nodeInfo.getRangeInfo();
        ^
  symbol:   class RangeInfoCompat
  location: class AccessibilityHierarchyDumper
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
13 errors

FAILURE: Build failed with an exception.

```

**Comment by xiphirx** [comment]
Yeah.. that seems to be the case. There should probably be a fallback check for older version compatibility. Something like 

```
hierarchy_data = get_view_hierarchy(output_dir, screenshot)
if KEY_VIEW_HIERARCHY in hierarchy_data:
  hierarchy = hierarchy_data[KEY_VIEW_HIERARCHY]
  ax_hierarchy = hierarchy_data[KEY_AX_HIERARCHY]
else:
  hierarchy = hierarchy_data
  ax_hierarchy = None
```

**Comment by xiphirx** [comment]
I'll fix the build issues

**Comment by xiphirx** [comment]
I have committed a bunch of fixes, things should work off master now.

**Comment by Laimiux** [comment]
I've updated the PR. I didn't test the changes with the latest master, what's the best way to confirm behavior? Do you do any checks?

**Comment by xiphirx** [comment]
You can run
```
$ gradle :plugin:pyTests
$ gradle :core:connectedAndroidTest
```

You can also run the screeenshot tests from the sample app

File: plugin/src/py/android_screenshot_tests/pull_screenshots.py [hunk 1]
```diff
@@ -566,7 +568,16 @@ def main(argv):
         puller_args += ["-s", opts["-s"]]
 
     multiple_devices = opts.get('--multiple-devices')
-    device_calculator = DeviceNameCalculator() if multiple_devices else NoOpDeviceNameCalculator()
+
+    device_calculator = None
+    if not should_perform_pull and multiple_devices:
+        device_name = opts.get('--device-name')
+        # TODO add validation that device_name is defined.
```
Comment (xiphirx) [inline plugin/src/py/android_screenshot_tests/pull_screenshots.py]: This should be done here. I would just exit + show usage.

**Comment by carlospaulino** [comment]
@Laimiux Are you planning to implement the changes proposed by @xiphirx ?

**Comment by Laimiux** [comment]
If anyone has the bandwidth to take this over, it would be great. I ended up using a different Gradle plugin so finishing this isn't a priority.

**Comment by radixdev** [comment]
@laimiux which different plugin did you end up using?

**Comment by Laimiux** [comment]
> @Laimiux which different plugin did you end up using?

@radixdev I've used https://github.com/Karumi/Shot

**Comment by oradkovsky** [comment]
Just in case someone finds helpful, I have a bit different take into this: https://github.com/oradkovsky/screenshot-tests-for-android/tree/customizations-0.14.3

**Comment by xiphirx** [comment]
I'm going to close this out due to lack of interest to push it forward. If someone wants to take this up please open a new PR, thanks!

**Pull request closed without merging**